src/UserSettingsPage.cpp

```diff
@@ -286,112 +239,87 @@ UserSettingsPage::UserSettingsPage(QSharedPointer<UserSettings> settings, QWidge
-        deviceFingerprintLayout->addWidget(deviceFingerprintLabel, 1);
-        deviceFingerprintLayout->addWidget(deviceFingerprintValue_);
 
-        auto sessionKeysLayout = new QHBoxLayout;
-        sessionKeysLayout->setContentsMargins(0, OptionMargin, 0, OptionMargin);
         auto sessionKeysLabel = new QLabel(tr("Session Keys"), this);
         sessionKeysLabel->setFont(font);
-        sessionKeysLayout->addWidget(sessionKeysLabel, 1);
+        sessionKeysLabel->setMargin(OptionMargin);
 
         auto sessionKeysImportBtn = new QPushButton{tr("IMPORT"), this};
-        connect(
-          sessionKeysImportBtn, &QPushButton::clicked, this, &UserSettingsPage::importSessionKeys);
         auto sessionKeysExportBtn = new QPushButton{tr("EXPORT"), this};
-        connect(
-          sessionKeysExportBtn, &QPushButton::clicked, this, &UserSettingsPage::exportSessionKeys);
+
+        auto sessionKeysLayout = new QHBoxLayout(this);
+        sessionKeysLayout->addWidget(new QLabel("", this), 1, Qt::AlignRight);
         sessionKeysLayout->addWidget(sessionKeysExportBtn, 0, Qt::AlignRight);
         sessionKeysLayout->addWidget(sessionKeysImportBtn, 0, Qt::AlignRight);
 
-        encryptionLayout_->addLayout(deviceIdLayout);
-        encryptionLayout_->addLayout(deviceFingerprintLayout);
-        encryptionLayout_->addWidget(new HorizontalLine{this});
-        encryptionLayout_->addLayout(sessionKeysLayout);
+        auto boxWrap = [this, &font](QString labelText, QWidget *field) {
+                auto label = new QLabel(labelText, this);
+                label->setFont(font);
+                label->setMargin(OptionMargin);
 
-        font.setWeight(QFont::Medium);
+                auto layout = new QHBoxLayout(this);
+                layout->addWidget(field, 0, Qt::AlignRight);
 
-        auto encryptionLabel_ = new QLabel(tr("ENCRYPTION"), this);
-        encryptionLabel_->setSizePolicy(QSizePolicy::Ignored, QSizePolicy::Fixed);
-        encryptionLabel_->setFont(font);
+                formLayout_->addRow(label, layout);
+        };
 
-        auto general_ = new QLabel(tr("GENERAL"), this);
-        general_->setSizePolicy(QSizePolicy::Ignored, QSizePolicy::Fixed);
-        general_->setFont(font);
+        formLayout_->addRow(general_);
+        formLayout_->addRow(new HorizontalLine(this));
+        boxWrap(tr("Minimize to tray"), trayToggle_);
+        boxWrap(tr("Start in tray"), startInTrayToggle_);
+        formLayout_->addRow(new HorizontalLine{this});
+        boxWrap(tr("Circular Avatars"), avatarCircles_);
+        boxWrap(tr("Group's sidebar"), groupViewToggle_);
+        boxWrap(tr("Typing notifications"), typingNotifications_);
+        formLayout_->addRow(new HorizontalLine{this});
+        boxWrap(tr("Read receipts"), readReceipts_);
+        boxWrap(tr("Send messages as Markdown"), markdownEnabled_);
+        boxWrap(tr("Desktop notifications"), desktopNotifications_);
+        formLayout_->addRow(new QLabel("", this));
```

Comment:
Are you using an empty label as a spacer here?

Reply:
ah yes. I tried to set a margin on the field below, but couldn't isolate the top/left/bottom/right values so the margin applied nicely on top but shoved the label in toward the middle the same amount.